internal/datastore/crdb/crdb_test.go

```diff
@@ -111,6 +123,69 @@ func TestCRDBDatastoreWithFollowerReads(t *testing.T) {
 	}
 }
 
+var defaultKeyForTesting = proxy.KeyConfig{
+	ID: "defaultfortest",
+	Bytes: (func() []byte {
+		b, err := hex.DecodeString("000102030405060708090A0B0C0D0E0FF0E0D0C0B0A090807060504030201000")
+		if err != nil {
+			panic(err)
+		}
+		return b
+	})(),
+	ExpiredAt: nil,
+}
+
+func TestCRDBDatastoreWithIntegrity(t *testing.T) {
+	b := testdatastore.RunCRDBForTesting(t, "")
+
+	test.All(t, test.DatastoreTesterFunc(func(revisionQuantization, gcInterval, gcWindow time.Duration, watchBufferLength uint16) (datastore.Datastore, error) {
+		ctx := context.Background()
+		ds := b.NewDatastore(t, func(engine, uri string) datastore.Datastore {
+			ds, err := NewCRDBDatastore(
+				ctx,
+				uri,
+				GCWindow(gcWindow),
+				RevisionQuantization(revisionQuantization),
+				WatchBufferLength(watchBufferLength),
+				OverlapStrategy(overlapStrategyPrefix),
+				DebugAnalyzeBeforeStatistics(),
+				WithIntegrity(true),
+			)
+			require.NoError(t, err)
+
+			wrapped, err := proxy.NewRelationshipIntegrityProxy(ds, defaultKeyForTesting, nil)
```

Comment:
Because it exists outside of CRDB's implementation; its a generic proxy that currently works with CRDB and Memdb (for testing)

Reply:
It just seems confusing that you have to do both - if you forget to add the proxy in, you will think you have integrity enabled but there will be no validation

Reply:
Well, no, you'd have integrity *data storage* enabled, but it would fail due to checks ensuring the data is actually present. Having it be in its own proxy file makes testing easier